src/modules/universal-discovery/components/create/choose.language.component.js

```diff
@@ -26,9 +27,23 @@ export default class ChooseLanguageComponent extends Component {
             selectedLanguageCode = props.allowedLanguages[0];
         }
 
+        if (selectedLanguageCode === null || !props.languages.mappings[selectedLanguageCode].enabled) {
+            const languages = Object.values(props.languages.mappings);
+            const language = languages.find((language) => {
+                const isAllowedLanguage = props.allowedLanguages.length ? props.allowedLanguages.includes(language.languageCode) : true;
+
+                return isAllowedLanguage && language.enabled;
+            });
+
+            selectedLanguage = language || {};
```

Comment:
I would set it only if the `language` exists.
Thanks to the changes `selectedLanguage` is never `null`